### PR TITLE
Compiling to no std

### PR DESCRIPTION
**Pull request by filiplazovic**
It a little chunky, but basically I added `rustfmt.toml` and ran `cargo fmt`. Let me know if you like the format config

### CODE REVIEW

File: gadgets/src/transaction/tests.rs [hunk 1]
```diff
@@ -83,31 +91,36 @@ fn test_is_valid_transaction_spend() {
 			let mut prover_transcript = Transcript::new(b"Transaction");
 			let mut prover = Prover::new(&pc_gens, &mut prover_transcript);
 
-			let (com_input_inverse_val, var_input_inverse_val) = prover.commit(input_inverse.clone(), Scalar::random(&mut test_rng));
+			let (com_input_inverse_val, var_input_inverse_val) =
```
Comment (drewstone) [inline gadgets/src/transaction/tests.rs:94]: would rather the args break the line so:
```
let (com_input_inverse_val, var_input_inverse_val) = prover.commit(
        input_inverse.clone(),
        Scalar::random(&mut test_rng)
);
```
possible?

File: node/src/service.rs [hunk 2]
```diff
@@ -244,7 +286,10 @@ pub fn new_light(mut config: Configuration) -> Result<TaskManager, ServiceError>
 	let (client, backend, keystore_container, mut task_manager, on_demand) =
 		sc_service::new_light_parts::<Block, RuntimeApi, Executor>(&config)?;
 
-	config.network.notifications_protocols.push(sc_finality_grandpa::GRANDPA_PROTOCOL_NAME.into());
+	config
```
Comment (drewstone) [inline node/src/service.rs:289]: Any way to make this not break instantly? Maybe at least after the first one?

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good, I think most of it is fine by me, but some of it is a bit clunky like newline for curly braces for function blocks and always breaking when there's like 2-3 arguments in a function. Can we set breakpoints to be at a character limit of 120/130?

File: gadgets/src/smt/smt.rs [hunk 3]
```diff
@@ -145,21 +185,89 @@ impl<'a> VanillaSparseMerkleTree<'a> {
...
+	}
+
+	pub fn prove_zk(
+		&self,
+		k: Scalar,
+		root: Scalar,
+		bp_gens: &BulletproofGens,
+		mut prover: Prover,
+	) -> (
+		R1CSProof,
```
[the remark this answers is not in the document]
Comment (filiplazovic) [reply gadgets/src/smt/smt.rs:178]: I've set the width to 120, but `rustfmt` has its own primitive ways of breaking the lines which I can't configure

File: node/src/service.rs (hunk 2 above)
Comment (filiplazovic) [reply node/src/service.rs:273]: Looks like it breaks all of them on a new line as soon as it exceeds the max

File: gadgets/src/transaction/tests.rs (hunk 1 above)
Comment (filiplazovic) [reply gadgets/src/transaction/tests.rs:85]: Looks like it's not. It first breaks the line after the `=` sign and then looks at the arguments

File: gadgets/src/smt/smt.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (drewstone) [reply gadgets/src/smt/smt.rs:178]: Is there any way we can turn some of it off?

**Comment by filiplazovic** [comment]
This is ready @drewstone

**Pull request merged**